File: LeetCode/Easy/0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        StringBuilder sb =  new StringBuilder();

        for (int i = 0; i < s.length(); i++) {
            if ((s.charAt(i) >= 'a' && s.charAt(i) <= 'z') 
            || (s.charAt(i) >= '0' && s.charAt(i) <= '9')){
                sb.append(s.charAt(i));
            }
        }

        String conv = sb.toString();
        int left = 0;
        int right = conv.length()-1;

        while (left < right) {
            if (conv.charAt(left) != conv.charAt(right)) {
                return false;
            }
            left += 1;
            right -= 1;
        }
        return true;
    }
}

File: LeetCode/Easy/0125-valid-palindrome/0125-valid-palindrome.go
func isPalindrome(s string) bool {
    n := len(s)
    var t []byte

    for i := 0; i < n; i++ {
        if (s[i] >= 'A' && s[i] <= 'Z') {
            t = append(t, 'a' + s[i] - 'A')
        } else if (s[i] >= 'a' && s[i] <= 'z') {
            t = append(t, s[i])
        } else if (s[i] >= '0' && s[i] <= '9') {
            t = append(t, s[i])
        }
    }

    k := len(t)

    for i := 0; i < k; i++ {
        if (t[i] != t[k-i-1]) {
            return false
        }
    }
    return true
}